lib/spack/spack/modules/common.py

```diff
@@ -623,7 +623,7 @@ def use_name(self):
         name = self.spec.format(projection)
         # Not everybody is working on linux...
         parts = name.split("/")
-        name = os.path.join(*parts)
+        name = "/".join(parts)
```

Comment:
@johnwparent Why are we doing this? I just want to note that with this change the operation is an identity:
```python
Python 3.8.10 (default, Mar 13 2023, 10:26:41) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> s = "/etc/foo/bar/baz"
>>> s.split("/")
['', 'etc', 'foo', 'bar', 'baz']
>>> "/".join(s.split("/"))
'/etc/foo/bar/baz'
>>> t = "/".join(s.split("/"))
>>> s == t
True

```

Reply:
This is @markus-ferrell's PR so I'll let him handle the details, but you're right, this isn't an exact parallel of the previous behavior/is an identity op. I assume this is necessary because some detail of the modules expect unix style path seps, but IMO a better approach would be to use pathlib's `as_posix` if that's necessary or to refactor whatever part of the module requires these if we have control over that.

Reply:
@johnwparent Sorry, my question was more to double check with you that changing a module name from e.g. `foo\1.2.3` to `foo/1.2.3` is still correct on Windows. Are module files supported on Windows at all?